lib/survey_gizmo/configuration.rb

```diff
@@ -110,8 +117,24 @@ def region=(region)
       region_infos = REGION_INFO[region]
       raise ArgumentError.new("Unknown region: #{region}") unless region_infos
 
-      @api_url = region_infos[:url]
+      @api_tld = region_infos[:tld]
       @api_time_zone = region_infos[:locale]
+
+      build_api_url
+    end
+
+    def api=(api)
+      raise ArgumentError.new("Unknown api: #{api}") unless APIS_INFO.keys.include?(api)
+
+      @api = APIS_INFO[api]
+
+      build_api_url
+    end
+
+    private
+
+    def build_api_url
```

Comment:
I extracted the `@api_url` instance variable assignation into a separate method because both `#region=` and `#api=` should rebuild `@api_url`